swarm/src/connection/pool.rs

```diff
@@ -692,17 +691,45 @@ where
                     let check_peer_id = || {
                         if let Some(peer) = expected_peer_id {
                             if peer != obtained_peer_id {
-                                return Err(PendingConnectionError::WrongPeerId {
-                                    obtained: obtained_peer_id,
-                                    endpoint: endpoint.clone(),
-                                });
+                                return match &endpoint {
+                                    ConnectedPoint::Dialer { address, .. } => {
+                                        Err(PoolEvent::PendingOutboundConnectionError {
+                                            id,
+                                            error: PendingOutboundConnectionError::WrongPeerId {
+                                                obtained: obtained_peer_id,
+                                                address: address.clone(),
+                                            },
+                                            peer: expected_peer_id.or(Some(obtained_peer_id)),
```

Comment:
```suggestion
                                            peer,
```

We can use `peer` directly since we are in the `if let Some(peer) = expected_peer_id` branch.

Reply:
We will still have to use `peer: Some(peer)` because of `peer: Option<PeerId>`

Reply:
Done.